### PR TITLE
feat: docker container action and compose

### PR DESCRIPTION
**Pull request by sebasptsch**
This adds a publish step on push to main. It also contains a demo docker-compose file

### CODE REVIEW

**Comment by sebasptsch** [comment]
woop

File: docker-compose.yaml
Comment (sunsetkookaburra) [inline docker-compose.yaml:1]: `/comp` again. Also are compose files in the repo normally examples or used directly? If it's more direct I'd probably say `./comp` the local folder would be better. Also not sure if composing the comp from the current directory would be better or more broadly applicable.
e.g. `.:/comp`

**Review by sunsetkookaburra — CHANGES_REQUESTED** [review]

**Comment by sunsetkookaburra** [comment]
just considering too now whether this is better suited to the ProgComp repository.
how hard would it be for someone to extend the container to host runtimes for the languages they wrote problems in?

**Comment by sunsetkookaburra** [comment]
Commenting for ref / to delete branch for now
```yaml
# .github/workflows/publish.yml
name: Publish Docker image
on:
  push:
    branches:
      - main

jobs:
  docker-build:
    permissions:
      contents: read
      packages: write
    runs-on: ubuntu-latest
    name: "Build and Publish Image"
    steps:
      - name: Checkout
        uses: actions/checkout@v4

      - name: Container Metadata
        id: container_metadata
        uses: docker/metadata-action@v5
        with:
          # list of Docker images to use as base name for tags
          images: |
            ghcr.io/progsoc/fuzzjudge
          # generate Docker tags based on the following events/attributes
          tags: |
            type=raw,value=latest
      - name: Set up QEMU
        uses: docker/setup-qemu-action@v3

      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v3

      - name: Login to GitHub Container Registry
        uses: docker/login-action@v3
        with:
          registry: ghcr.io
          username: ${{ github.repository_owner }}
          password: ${{ secrets.GITHUB_TOKEN }}
      - name: Build and push (Backend)
        uses: docker/build-push-action@v5
        id: docker_build_backend
        with:
          tags: ${{ steps.container_metadata.outputs.tags }}
          push: true
          labels: ${{ steps.container_metadata.outputs.labels }}
          cache-from: type=registry,ref=ghcr.io/progsoc/fuzzjudge:latest
```
```dockerfile
# Dockerfile
FROM denoland/deno

WORKDIR /app
COPY ./src ./src

EXPOSE 8000
CMD ["run", "-A", "src/main.ts", "/competition"]
```
```yaml
# docker-compose.yml
services:
  server:
    build: .
    volumes:
      - ./sample:/competition
      - type: bind
        source: ./db.sqlite3
        target: /app/scores.db
    ports:
      - "8000:8000"
```

**Pull request closed without merging**